features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Comments/CommentAdd.jsx

```diff
@@ -128,14 +133,13 @@ class CommentAdd extends React.Component {
                 .then((newComment) => {
                     this.setState({ content: '' });
                     const addedComment = newComment.body;
-                    // if (parentCommentId === null) {
-                    //     allComments.push(addedComment);
-                    // } else {
-                    //     const index = allComments.findIndex(this.filterCommentToAddReply);
-                    //     allComments[index].replies.push(addedComment);
-                    //     toggleShowReply();
-                    // }
-                    allComments.push(addedComment);
+                    if (parentCommentId === null) {
+                        allComments.push(addedComment);
```

Comment:
We are directly pushing items to parent's state here, this should do in a callback method something like `onAddComment`